plugin/trino-pulsar/src/main/java/io/trino/plugin/pulsar/decoder/avro/PulsarAvroRowDecoder.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pulsar.decoder.avro;
+
+import io.netty.buffer.ByteBuf;
+import io.trino.decoder.DecoderColumnHandle;
+import io.trino.decoder.FieldValueProvider;
+import io.trino.plugin.pulsar.PulsarRowDecoder;
+import io.trino.spi.TrinoException;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.pulsar.client.impl.schema.generic.GenericAvroRecord;
+import org.apache.pulsar.client.impl.schema.generic.GenericAvroSchema;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Functions.identity;
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static io.trino.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Refer to {@link io.trino.decoder.raw.RawRowDecoder}.
+ */
+public class PulsarAvroRowDecoder
+        implements PulsarRowDecoder
+{
+    private final GenericAvroSchema genericAvroSchema;
+    private final Map<DecoderColumnHandle, PulsarAvroColumnDecoder> columnDecoders;
+
+    public PulsarAvroRowDecoder(GenericAvroSchema genericAvroSchema,
+                                Set<DecoderColumnHandle> columns)
+    {
+        this.genericAvroSchema = requireNonNull(genericAvroSchema, "genericAvroSchema is null");
+        columnDecoders = columns.stream()
+                .collect(toImmutableMap(identity(), this::createColumnDecoder));
+    }
+
+    private PulsarAvroColumnDecoder createColumnDecoder(DecoderColumnHandle columnHandle)
+    {
+        return new PulsarAvroColumnDecoder(columnHandle);
+    }
+
+    /**
+     * Decode ByteBuf by {@link org.apache.pulsar.client.api.schema.GenericSchema}.
+     *
+     * @param byteBuf
+     * @return
+     */
+    @Override
+    public Optional<Map<DecoderColumnHandle, FieldValueProvider>> decodeRow(ByteBuf byteBuf)
+    {
+        GenericRecord avroRecord;
+        try {
+            GenericAvroRecord record = (GenericAvroRecord) genericAvroSchema.decode(byteBuf);
+            avroRecord = record.getAvroRecord();
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Please remove. And any other occurences of `printStackTrace` from production code.

Reply:
removede